### DISCUSSION TITLE
Deleting pages from the WebUI

### DISCUSSION THREAD

**Discussion by PhilippKammann** [gotson/komga#1590] [Ideas]
### **Problem**
Some scanlation groups put the chapter number in the scanner page and these pages won't be recognized as duplicate pages and therefore not automatically deleted.
The scanner pages replace the actual thumbnail and/or shift the page numbering and break up two-pages spreads in the double-pages layout.

### **Solution I would like**
The ability to delete pages directly from the WebUI (as an admin) 

### **My current solution**
Going through the files on my server and deleting the page manually from the archive.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by profzelonka** [comment]
A good simple implementation could look in the same form of the current ability to "Download current page" when reading a comic and clicking on the three dots in the top corner, just "Delete current page". A more complex one would be a neat "Mark to delete page" option which would queue pages for a cron to complete at scheduled time, kind of like the current functionality of deleting duplicate pages.
(Love the app, thanks for all the work on it.)

**Comment by trx1138** [comment]
> My current solution
> Going through the files on my server and deleting the page manually from the archive.

I've been doing the same painful process and came by to suggest another duplicate, manual page deletion.
in my case, I'm running automated archiving from various sources with quite complicated filters but sometimes there's still unwanted pages included.
unfortunately, post process for this kind of unwanted pages can not be automated and needs manual intervention.
moreover, due to security reasons and etc, I have to do this all by myself even if I'm not interested in a certain series. :(
so a new user privilege to delete or mark to delete unwanted will be great addition, especially if one's running automated archiving.

really hope #800 and its duplicates could be accepted.

**Comment by gotson** [comment]
@PhilippKammann would you want to only delete the page, or mark the page as a duplicate somehow so it shows in the Duplicate Pages ?

Deleting the page would be limited to some books for which repackaging can be done, most likely rar4, cbz, maybe PDF. Epub that are divina compatible would be more complex as it would require to repack the epub with a modified manifest.

Is the page usually at the beginning or end of the book ? Deleting the page could potentially trigger a book update event, which would instruct the webui to reload the book, which could disrupt the reading experience.

Also wanted to check whether you are interested in actually deleting the image inside the archive, or mark the page as deleted and somehow not show it (whether in the API or other medium) ?

**Reply by PhilippKammann** [reply]
Sorry, Github didn't notify me there was an update on this discussion.

- I would like to delete the page. Marking it as duplicate wouldn't really matter, if it is for one-off unique pages (like chapter pages from scanlators)
- I don't think it would be necessary to do that on-the-fly, closing the book, scanning the library and reopening would be okay. Which would mean marking it as a "Duplicate Page (auto delete)" would work here and not disrupt the reading process
- I don't really see the benefit of having the page remain in the file, if I want it gone. So it should be deleted.

**Comment by SytheZN** [comment]
Wouldn't it work to do what's done with the current implementation of duplicate pages?

In my mind, if [List book pages](https://komga.org/docs/openapi/get-book-pages) included `pageHash` and `url` then you'd have all the info required to call [Delete specific duplicate page](https://komga.org/docs/openapi/delete-single-match) with an arbitrary page.

On the UI side I picture a trashcan icon on the page thumbnail in the reader page list.

**Reply by SytheZN** [reply]
@gotson I know you said you're not comfortable merging changes with many moving pieces, but there's a working implementation [here](https://github.com/SytheZN/komga/tree/feat/delete-arbitrary-pages) if you'd like to take a look.

High level, I added an endpoint to the hash-pages controller to hash and then delete arbitrary pages using the existing logic. Less moving parts than I expected following our last discussion.

**Comment by Cookies77777** [comment]
I would also like to see this feature officially added, especially for cbz format.
Three dots and "Delete current page" or marking a page as duplicate manually for deletion later.
Right now a lot of pages don't get marked as duplicate as mention above with chapter number changing and etc.
I usually fix the chapters manually and it's tedious and sometimes I just give up.
Rename cbz to zip, extract, delete page, rezip, rename back to cbz, replace file.
If it was built in and if I could just delete the page during reading then that would be really really helpful.
